android-lib/src/androidTest/java/com/kinvey/androidTest/store/DataStoreTest.java

```diff
@@ -2449,4 +2449,46 @@ public void tearDown() {
         }
     }
 
+    @Test
+    public void testDeltaCache() throws InterruptedException, IOException {
+        client.setUseDeltaCache(true);
+        DataStore<Person> store = DataStore.collection(Person.COLLECTION, Person.class, StoreType.SYNC, client);
+        assertTrue(store.isDeltaSetCachingEnabled());
+        client.getSyncManager().clear(Person.COLLECTION);
+        clearBackend(store);
+
+        Person person = new Person();
+        person.setUsername("name");
+        person.set("CustomField", "CustomValue");
+        store.save(person);
+
+        store.syncBlocking(null);
+        Person person2 = new Person();
+        person2.setUsername("changed name 1");
+        person2.set("CustomField", "CustomValueChanged ");
+        store.save(person2);
+
+        Person person3 = new Person();
+        person3.setUsername("changed name 2");
+        person3.set("CustomField", "CustomValueChanged 2");
+        store.save(person3);
+
+        DefaultKinveySyncCallback callback = sync(store, DEFAULT_TIMEOUT);
+        assertNull(callback.error);
+        assertNotNull(callback.kinveyPushResponse);
+        assertNotNull(callback.kinveyPullResponse);
+
+        List<Person> personList = store.find();
+        Person person1 = personList.get(0);
+        assertNotNull(person1);
+        assertNotNull(person1.getUsername());
+    }
+
+    @Test
+    public void testDeltaCacheAfterDataStoreInitialization() throws InterruptedException, IOException {
```

Comment:
This test describes how the SDK behaves today, but I think we should improve the behavior.

I recommend logging the following as a separate enhancement -
The `setUseDeltaCache` flag should be on the datastore instance. It should be set after the datastore is created.

For this ticket, it is ok to leave the test in.

Reply:
It makes sense! I'll create a ticket for this.